docs/architecture/maui/dependency-injection.md

```diff
@@ -1,9 +1,192 @@
+
+There are many dependency injection containers available, with the eShopOnContainers mobile app using Microsoft.Extensions.DependencyInjection to manage the instantiation of views, view models, and service classes in the app. Autofac facilitates building loosely coupled apps, and provides all of the features commonly found in dependency injection containers, including methods to register type mappings and object instances, resolve objects, manage object lifetimes, and inject dependent objects into constructors of objects that it resolves. For more information about Microsoft.Extensions.DependencyInjection, see [Microsoft.Extensions.DependencyInjection](/dotnet/core/extensions/dependency-injection) on Microsoft Developer Center.
+
+In .NET MAUI, the `MauiProgram` class will call into the `CreateMauiApp` method to create a `MauiAppBuilder` object. The `MauiAppBuilder` object has a `Services` property of type `IServiceCollection` which provides place to register our components, such as views, view models and services, for dependency injection. Any components registered with the `Services` property will be provided to the dependency injection container when the `MauiAppBuilder.Build` method is called.
+
+At runtime, the container must know which implementation of the services are being requested in order to instantiate them for the requested objects. In the eShopOnContainersMobile app, the `IAppEnvironmentService`, `IDialogService` , `INavigationService`, and `ISettingsService` interfaces need to be resolved before it can instantiate a `ProfileViewModel` object. This involves:
+
+- The container deciding how to instantiate an object that implements the interface. This is known as *registration*.
+- The container instantiating the object that implements the required interface, and the `ProfileViewModel` object. This is known as *resolution*.
+
+Eventually, the app will finish using the `ProfileViewModel` object and it will become available for garbage collection. At this point, the garbage collector should dispose of any short-lived interface implementations, if other classes do not share the same instance.
+
+## Registration
+
+Before dependencies can be injected into an object, the types of the dependencies must first be registered with the container. Registering a type involves passing the container an interface and a concrete type that implements the interface.
+
+There are two ways of registering types and objects in the container through code:
+
+- Register a type or mapping with the container. This is known as transient registration. When required, the container will build an instance of the specified type.
+- Register an existing object in the container as a singleton. When required, the container will return a reference to the existing object.
+
+> [!TIP]
+> Dependency injection containers are not always suitable
```

Comment:
```suggestion
> Dependency injection containers are not always suitable.
```

Why? I wasn't able to suggest a change here besides punctuation, but I'm not sure when DI is not suitable, please elaborate here. Also, please try to remember alerts require complete sentences and punctuation.